code/modules/clothing/shoes/magboots.dm

```diff
@@ -26,45 +26,46 @@
 	. = ..()
 	if(slot != SLOT_HUD_SHOES || !ishuman(user))
 		return
-	check_mag_pulse()
+	check_mag_pulse(user)
 
 /obj/item/clothing/shoes/magboots/dropped(mob/user, silent)
 	. = ..()
 	if(!ishuman(user))
 		return
-	check_mag_pulse()
+	check_mag_pulse(user, removing = TRUE)
 
-/obj/item/clothing/shoes/magboots/attack_self(mob/user, forced = FALSE)
-	toggle_magpulse(user, forced)
+/obj/item/clothing/shoes/magboots/attack_self(mob/user)
+	toggle_magpulse(user)
 
-/obj/item/clothing/shoes/magboots/proc/toggle_magpulse(mob/user, forced)
+/obj/item/clothing/shoes/magboots/proc/toggle_magpulse(mob/user, no_message)
 	if(magpulse) //magpulse and no_slip will always be the same value unless VV happens
 		REMOVE_TRAIT(user, TRAIT_NOSLIP, UID())
 		slowdown = slowdown_passive
 	else
-		ADD_TRAIT(user, TRAIT_NOSLIP, UID())
+		if(user.get_item_by_slot(SLOT_HUD_SHOES) == src)
+			ADD_TRAIT(user, TRAIT_NOSLIP, UID())
 		slowdown = slowdown_active
 	magpulse = !magpulse
 	no_slip = !no_slip
 	if(multiple_icons)
 		icon_state = "[magboot_state][magpulse]"
-	if(!forced)
+	if(!no_message)
 		to_chat(user, "You [magpulse ? "enable" : "disable"] the [magpulse_name].")
 	user.update_inv_shoes()	//so our mob-overlays update
 	user.update_gravity(user.mob_has_gravity())
 	for(var/X in actions)
 		var/datum/action/A = X
 		A.UpdateButtons()
-	check_mag_pulse(user)
+	check_mag_pulse(user, removing = (user.get_item_by_slot(SLOT_HUD_SHOES) != src))
 
-/obj/item/clothing/shoes/magboots/proc/check_mag_pulse(mob/user)
+/obj/item/clothing/shoes/magboots/proc/check_mag_pulse(mob/user, removing = FALSE)
 	if(!user)
 		return
-	if(magpulse)
-		ADD_TRAIT(user, TRAIT_MAGPULSE, "magboots")
+	if(magpulse && !removing)
+		ADD_TRAIT(user, TRAIT_MAGPULSE, "magboots[UID()]")
 		return
 	if(HAS_TRAIT(user, TRAIT_MAGPULSE)) // User has trait and the magboots were turned off, remove trait
-		REMOVE_TRAIT(user, TRAIT_MAGPULSE, "magboots")
+		REMOVE_TRAIT(user, TRAIT_MAGPULSE, "magboots[UID()]")
```

Comment:
I think changing these both to `UNIQUE_TRAIT_SOURCE` could be a good idea

Reply:
`#define UNIQUE_TRAIT_SOURCE(target) "unique_source_[UID(target)]"`
This define doesn't even work properly, and I think its better to have more information where its actually coming from so people debugging don't have to find the UID to know what is providing the trait

Reply:
PR is up to fix it, we probably don't need to enforce it to always be used but a UID should always be used